src/Lean/Meta/GetUnfoldableConst.lean

```diff
@@ -39,7 +39,10 @@ def getUnfoldableConst? (constName : Name) : MetaM (Option ConstantInfo) := do
   match (← getEnv).find? constName with
   | some (info@(.thmInfo _))  => getTheoremInfo info
   | some (info@(.defnInfo _)) => if (← canUnfold info) then return info else return none
-  | some info                 => return some info
+  | some (info@(.recInfo _))  => return some info
```

Comment:
The name of this function is strange. It is currently called `getUnfoldableConst?` but some of the cases do not even have something to unfold (e.g., `recInfo`). 
I found two use cases for this function: 
- `isDeltaCandidate?`: This one immediately checks whether the `info` has a value or not. We can optimize this for this case and have clean code.
- `matchConstAux`: which is used to handle orthogonal cases using the same API. We should cleanup this one. This function looks odd to me.

I prefer to cleanup and ensure the function names make sense. If we decide to not cleanup, we should add comments explaining why the function implementation is counterintuitive.